src/bmp/decoder.rs

```diff
@@ -1255,8 +1254,24 @@ impl<R: Read + Seek> BMPDecoder<R> {
     }
 }
 
+/// Wrapper struct around a `Cursor<Vec<u8>>`
+pub struct BmpReader(Cursor<Vec<u8>>);
```

Comment:
What about doing something like:
```rust
pub struct BmpReader<R> {
    inner: Cursor<Vec<u8>>,
    _phantom: PhantomData<R>,
}
```
Because right now when we update these reader types to take the inner reader as an argument it will still be a breaking change.